eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/MutableList.java

```diff
@@ -60,6 +60,26 @@
 public interface MutableList<T>
         extends MutableCollection<T>, List<T>, Cloneable, ListIterable<T>
 {
+    /**
+     * This default override exists because java.util.List added a default getFirst() method in Java 21.
+     * @since 12.0
+     */
+    @Override
+    default T getFirst()
+    {
+        return this.isEmpty() ? null : this.get(0);
+    }
+
+    /**
+     * This default override exists because java.util.List added a default getLast() method in Java 21.
+     * @since 12.0
+     */
+    @Override
+    default T getLast()
+    {
+        return this.isEmpty() ? null : this.get(this.size() - 1);
+    }
+
```

Comment:
> This is my first time looking at the changes coming in the SDK in version 21. The new getFirst() and getLast() methods have a different contract than ours. They throw, rather than return null, on empty collections. This is arguably a better choice, but it's a breaking change that we would never make if we didn't have to. I know this code review is focused on compiler errors and I'm not suggesting any code changes. Just wanted to start the conversation.
> 
> ```
> /**
>  * {@inheritDoc}
>  *
>  * @implSpec
>  * If this List is not empty, the implementation in this interface returns the result
>  * of calling {@code get(0)}. Otherwise, it throws {@code NoSuchElementException}.
>  *
>  * @throws NoSuchElementException {@inheritDoc}
>  * @since 21
>  */
> ```

I discovered this is the behavior for `MutableSortedSet` because it delegates to `first` / `last` methods which also throw `NoSuchElementException`. Agreed, we probably will never change it.

Reply:
I'm not advocating for any changes, just thinking out loud with you since this is all new to me. Just want to point out that `public interface List<E> extends SequencedCollection<E>` in Java 21, which means that LinkedList is a SequencedCollection that's not random access. That means the default implementation _in the JDK_ is suboptimal, but we're used to that.